### DISCUSSION TITLE
Disappearing checkbox NTP time synchronization

### DISCUSSION THREAD

**Discussion by GermanAizek** [lxqt/lxqt#2609] [Q&A]
![изображение](https://github.com/user-attachments/assets/68592970-c83f-4907-8ca9-e4fcea25dfe5)

Why is there checkbox on LUbuntu LxQT for time synchronization using NTP, but with clean installation in Debian there is no such checkbox?

My Debian 12:
![изображение](https://github.com/user-attachments/assets/957263f3-0d5d-480c-9b38-968fb9c1db8f)

```
LXQt Desktop Toolbox - Technical Info<p>
  LXQt About Version:     1.3.0
  LXQt Version:           1.3.0
  Qt:                     5.15.13
  Build type:             Release
  System Configuration:   /etc/xdg
  Share Directory:        /usr/local/share/lxqt
  Translations:           /home/debian/.local/share/lxqt/translations
/usr/local/share/lxqt/translations
/usr/share/lxqt/translations


User Directories
  Xdg Data Home:          /home/debian/.local/share
  Xdg Config Home:        /home/debian/.config
  Xdg Data Dirs:          /home/debian/.local/share
/usr/local/share
/usr/share
  Xdg Cache Home:         /home/debian/.cache
  Xdg Runtime Home:       /run/user/1000
  Xdg Autostart Dirs:     /etc/autostart
/etc/xdg/autostart
/usr/share/autostart
  Xdg Autostart Home:     /home/debian/.config/autostart
```

**Comment by stefonarch** [comment]
No idea, there were no relevant changes after v1.2 for `lxqt-admin-time`. Is ntp or  systemd-timesyncd installed?
On arch and debian testing there is no ntp anymore but  `systemd-timesyncd.service` by default.

**Comment by ashugg** [comment]
I am using Lubuntu 24.04 LTS and can confirm that I see the "Enable network time synchronisation (NTP)" checkbox as well, ticked by default.

Which Lubuntu release are you comparing with, GermanAizek? Lubuntu 24.04 LTS ships with LXQt 1.4.0; the previous release, Lubuntu 23.10 shipped with LXQt 1.3.0.

My best guess is that there was some reason at build time for the Debian 12 package that caused NO_SYSTEMD to be defined, because that will:

* use `NOSDTimeDateCtrl` instead of `DBusTimeDateCtrl`, where `useNtp()` and `setUseNtp()` both return false
* make `DateTimePage::reload()` set the `ui->ntp` control to not visible

I'd say you'd have to ask the Debian package maintainers about this.

I looked at [the amd64 buildd log](https://buildd.debian.org/status/fetch.php?pkg=lxqt-admin&arch=amd64&ver=1.2.0-1&stamp=1671034017&raw=0) for [lxqt-admin 1.2.0](https://packages.debian.org/stable/lxqt-admin). You can see early in the log that the build environment definitely installs systemd.

The cmake output doesn't tell us whether or not CMake thinks libsystemd is available, as far as I can tell. The only check is in `lxqt-admin-time/CMakeLists.txt` where NO_SYSTEMD is defined, and removing the QUIET keyword would help here.

Also, the linking of the `lxqt-admin-time` executable (search the log for "Linking CXX executable lxqt-admin-time", for me it's lines 2257-2259) shows both `dbustimedatectl.cpp.o` and `nosdtimedatectl.cpp.o` being passed to the linker, whereas with that build condition I'd have expected CMake to have built and link only one or the other. Maybe something more is needed in `CMakeLists.txt` to make that happen? Not something I know how to do, sorry.

Separately to all this, I think the tool needs a bit of a rethink to see if what it's trying to achieve can be done in a better way now. It's 8 years since PCMan wrote it and I think subsequent little tweaks have backed it rather awkwardly into a corner.

**Reply by GermanAizek** [reply]
@ashugg I have Debian Unstable-Testing (dev branch), this branch contains newest packages and they are newer than Debian 12 release.

```
debian@debian:~$ lxqt-admin-time -v
lxqt-admin-time 1.3.0
liblxqt   1.3.0
Qt        5.15.13
```

**Reply by ashugg** [reply]
Hi @GermanAizek, you initially said "My Debian 12" with the screenshot that has no "NTP" checkbox. Now, you are saying that you in fact are not running Debian 12?

I cannot work out which release of Debian you are running - unless it's disconnected from the internet, and is quite out of date. Debian testing (trixie) and unstable (sid) both have LXqt version 1.4.0 and Qt 5.15.15, and Debian experimental (actual dev) has LXqt version 2.0.0, relying on Qt 6.7.4 from unstable.

As you are showing LXQt version 1.3.0, it does not make sense - unless you built the LXQt packages yourself. With Debian packages, you should be on version 1.4.0 now. I think your system is in strange territory. No-one can reasonably assist you while it is in a state that doesn't match Debian's releases.
